app/templates/components/medication-allergy.hbs

```diff
@@ -0,0 +1,53 @@
+<label for="">{{t 'allergies.patientAllergy'}}</label>
+
+
+<ul class="list-inline allergy-list">
+  <li>
+    <button class="btn muted test-action" {{action "createNewAllergy"}}>{{t 'buttons.addNewAllergy'}}</button>
+  </li>
+  {{#each patient.allergies as |allergy index|}}
+    {{#if showAllAllergies}}
+      <li>
+        <button class="btn-link allergy-button" {{action "editAllergy" allergy}}>{{allergy.name}}</button>
+      </li>
+    {{else}}
+      {{#if (lte index 4)}}
+        <li>
+          <button class="btn-link allergy-button" {{action "editAllergy" allergy}}>{{allergy.name}}</button>
+        </li>
+      {{/if}}
+    {{/if}}
+  {{/each}}
+    {{#if (gt patient.allergies.length 4)}}
```

Comment:
This line should be 
```hbs
{{#if (gt patient.allergies.length 5)}}
```
because right now the showAll link displays even when there are no more allergies to display.